modules/cnf-topology-aware-lifecycle-manager-backup-recovery.adoc

```diff
@@ -0,0 +1,100 @@
+// Module included in the following assemblies:
+// Epic CNF-3901 (CNF-2133) (4.11), Story TELCODOCS-339
+// * scalability_and_performance/cnf-talm-for-cluster-upgrades.adoc
+
+:_content-type: PROCEDURE
+[id="talo-backup-recovery_{context}"]
+= Recovering after a failed upgrade
+
+If an upgrade fails, you can use the backup to return the system to its pre-upgrade state. There are two stages:
+
+Rollback:: If the attempted upgrade included a change to the platform OS deployment, it will be necessary to rollback to the previous version before running the recovery script.
+Recovery:: The recovery shuts down containers, and restores files from the backup partition to the targeted directories. A reboot is necessary. The second stage restores cluster database and related files from the backup, relaunches containers, and triggers redeployments to restore clusters.
+
+.Prerequisites
+
+* Install the {cgu-operator-first}.
+* Provision one or more managed clusters.
+* Install Advanced Cluster Management
+* Log in as a user with `cluster-admin` privileges.
+* Run an upgrade that is configured for backup.
+
+.Procedure
+
+. Check the status of the upgrade by running the following command:
++
+[source,terminal]
+----
+$ oc ostree admin status
+----
+.Example outputs
++
+[source,terminal]
+----
+[root@lab-test-spoke2-node-0 core]# ostree admin status
+* rhcos c038a8f08458bbed83a77ece033ad3c55597e3f64edad66ea12fda18cbdceaf9.0
+    Version: 49.84.202202230006-0
+    Pinned: yes <1>
+    origin refspec: c038a8f08458bbed83a77ece033ad3c55597e3f64edad66ea12fda18cbdceaf9
+----
+<1> The current deployment is pinned. A platform OS deployment rollback is not necessary.
++
+[source,terminal]
+----
+[root@lab-test-spoke2-node-0 core]# ostree admin status
+* rhcos f750ff26f2d5550930ccbe17af61af47daafc8018cd9944f2a3a6269af26b0fa.0
+    Version: 410.84.202204050541-0
+    origin refspec: f750ff26f2d5550930ccbe17af61af47daafc8018cd9944f2a3a6269af26b0fa
+rhcos ad8f159f9dc4ea7e773fd9604c9a16be0fe9b266ae800ac8470f63abc39b52ca.0 (rollback) <1>
+    Version: 410.84.202203290245-0
+    Pinned: yes <2>
+    origin refspec: ad8f159f9dc4ea7e773fd9604c9a16be0fe9b266ae800ac8470f63abc39b52ca
+----
+<1> This platform OS deployment is marked for rollback.
+<2> The previous deployment has been pinned and can be rolled back.
+. If you need to trigger a rollback of the platform OS deployment, run the following command:
++
+[source,terminal]
+----
+$ rpm-ostree rollback -r
+----
+. The first phase of the recovery shuts down containers, cleans up, and restores files from the backup partition to the targeted directories. To begin the recovery run the following command:
```

Comment:
This line seem redundant to line `12`, may be line 12 and 13 can be rephrased to make it generic?